OlenaNehodenko/HW3/3_1.py

```diff
@@ -0,0 +1,61 @@
+#First part in HW3 is to write Python's philosophy in the string format
+#find The number of occurrences of the some words
+my_string='''\nBeautiful is better than ugly.
+Explicit is better than implicit.
+Simple is better than complex.
+Complex is better than complicated.
+Flat is better than nested.
+Sparse is better than dense.
+Readability counts.
+Special cases aren't special enough to break the rules.
+Although practicality beats purity.
+Errors should never pass silently.
+Unless explicitly silenced.
+In the face of ambiguity, refuse the temptation to guess.
+There should be one and preferably only one obvious way to do it.
+Although that way may not be obvious at first unless you're Dutch.
+Now is better than never.
+Although never is often better than right now.
+If the implementation is hard to explain, it's a bad idea.
+If the implementation is easy to explain, it may be a good idea.
+Namespaces are one honking great idea let's do more of those'''
+print("The number of occurrences of the words 'better'=",my_string.find('better'))
+print("The number of occurrences of the words 'never'=",my_string.find('never'))
+print("The number of occurrences of the words 'is'=",my_string.find('is'))
+#display all text in uppercase (all letters in uppercase)
+print("Python's philosophy in uppercase:",my_string.upper())
+#replace all occurrences of the symbol “i” with “&”
+print("Python's philosophy with “&”:",my_string.replace('i','&'))
+
+#Second part in HW3 is work with a four-digit natural number
+#find the product of the digits of this number
```

Comment:
use more meaningful names for variables

number_for_product = int(input("Enter a four-digit natural number: "))
product_of_digits = 1
while number_for_product > 0:
    digit = number_for_product % 10
    product_of_digits *= digit
    number_for_product //= 10
print("Product of digits:", product_of_digits)


number_for_reverse = int(input("Enter a four-digit natural number: "))
reversed_number = 0
while number_for_reverse > 0:
    digit = number_for_reverse % 10
    reversed_number = reversed_number * 10 + digit
    number_for_reverse //= 10
print("Reverse number:", reversed_number)

number_for_sort = int(input("Enter a four-digit natural number: "))
sorted_number = int(''.join(sorted(str(number_for_sort))))
print("Ascending sort version of number:", sorted_number)